### PR TITLE
[PATCH v4] example: ml: add ML inference example

### PR DESCRIPTION
**Pull request by JereLeppanen**

### CODE REVIEW

File: platform/linux-generic/example/ml/conv_gen.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Nokia
+#
+
+import onnx
+from onnx import helper
+from onnx import TensorProto
```
Comment (TuomasTaipale) [inline platform/linux-generic/example/ml/conv_gen.py:7]: `README` should mention that `onnx` python library is needed.

File: platform/linux-generic/example/ml/conv_gen.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (TuomasTaipale) [inline platform/linux-generic/example/ml/conv_gen.py:1]: With python shebang (and executable rights), this wouldn't need `python3 conv_gen.py`.

File: example/ml/Makefile.am [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+			fi \
+		done \
+	fi
+
+clean-local:
+	if [ "x$(srcdir)" != "x$(builddir)" ]; then \
+		for f in $(EXTRA_DIST); do \
+			rm -f $(builddir)/$$f; \
+		done \
+	fi
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply example/ml/Makefile.am]: Removed.

File: example/ml/odp_ml_run.c [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+	if (!model_param.model) {
+		ODPH_ERR("Failed to read model file\n");
+		ret = -1;
+		goto odp_term;
+	}
+
+	glb.mdl = odp_ml_model_create(glb.opt.model_name, &model_param);
+	free(model_param.model);
+	if (glb.mdl == ODP_ML_MODEL_INVALID) {
+		ODPH_ERR("odp_ml_model_create failed.\n");
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply example/ml/odp_ml_run.c]: Missing "()" added, and periods removed.

File: platform/linux-generic/example/ml/conv_gen.py (hunk 2 above)
Comment (JereLeppanen) [reply platform/linux-generic/example/ml/conv_gen.py:1]: There are other similar scripts, which should all be changed if we want to change them. I'm leaving this as-is.

**Comment by JereLeppanen** [comment]
All comments processed in v3.

**Review by TuomasTaipale — APPROVED** [review]

**Comment by JereLeppanen** [comment]
v4:
- Rebase.
- Add review tags.

**Pull request merged**